chicken-sloths/bangazon-site-sprint3
# Editing Customer Data

# Description
This ticket allows users to edit their address and phone number. It also adds client-side and server-side validation to registering a customer and updating customer info.

## Related Ticket(s)
Fixes #78,
Fixes #9 

## Problem to Solve
Our customer faker data includes address and phone number. When we register new customers, we don't ask for their address or phone number. This builds out the functionality for customers to add and edit their address and phone number after they've registered.

## Proposed Changes
- Break out settings routes into a separate module
- Added basic validation for registration form
- Added basic validation on edit form
- Added route, view, and functionality to update customer data


## Expected Behavior && Steps to Test

1. Run `npm run db:gen`
1. Run `npm start`

### Test with a new user
1. Register a new user
1. Try entering an email address that's not formatted correctly. It shouldn't let you do it, and it should throw a helpful error message!
1. Once you've registered a new user, go into their account settings. Their account info should be blank
1. Go find that customer's info in your db browser by running the query: ``SELECT * FROM "Customers" WHERE "id"=51``. You should see null values for address, etc.
1. Click `Edit Your Information`
1. You should see a form with any customer information you've already entered (i.e. name) as default values and placeholder text explaining the rest of the fields.
1. Enter whatever information you want. 
1. It should prevent you from entering a zip code that's more than five digits.
1. Hit submit and go back to refresh your db browser! You should see the values you just updated.

### Test with an old user
1. Log in with any user
1. Bring their information up in your db browser by querying for their id
1. Update their information
1. When you refresh your query in the db browser, you should see the new info
